### PR TITLE
Convert SAM drivers to DT_INST

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
Moved the commits related to SAM into one PR, so easier to get review that way.

**Comment by galak** [comment]
TODO:
- [ ] drivers/usb/device/usb_dc_sam.c
- [ ] drivers/serial/usart_sam.c
- [ ] drivers/pinmux/pinmux_sam0.c
- [ ] drivers/gpio/gpio_sam0.c
- [ ] drivers/gpio/gpio_sam.c
- [ ] drivers/adc/adc_sam_afec.c
- [ ] drivers/adc/adc_sam0.c

File: soc/arm/atmel_sam/sam4e/dts_fixup.h [hunk 1]
```diff
@@ -67,19 +67,4 @@
...
-#define DT_USART_SAM_PORT_0_NAME             DT_ATMEL_SAM_USART_400A0000_LABEL
-#define DT_USART_SAM_PORT_0_BAUD_RATE        DT_ATMEL_SAM_USART_400A0000_CURRENT_SPEED
-#define DT_USART_SAM_PORT_0_IRQ              DT_ATMEL_SAM_USART_400A0000_IRQ_0
-#define DT_USART_SAM_PORT_0_IRQ_PRIO         DT_ATMEL_SAM_USART_400A0000_IRQ_0_PRIORITY
-#define DT_USART_SAM_PORT_0_PERIPHERAL_ID    DT_ATMEL_SAM_USART_400A0000_PERIPHERAL_ID
-#define DT_USART_SAM_PORT_1_NAME             DT_ATMEL_SAM_USART_400A4000_LABEL
-#define DT_USART_SAM_PORT_1_BAUD_RATE        DT_ATMEL_SAM_USART_400A4000_CURRENT_SPEED
-#define DT_USART_SAM_PORT_1_IRQ              DT_ATMEL_SAM_USART_400A4000_IRQ_0
-#define DT_USART_SAM_PORT_1_IRQ_PRIO         DT_ATMEL_SAM_USART_400A4000_IRQ_0_PRIORITY
-#define DT_USART_SAM_PORT_1_PERIPHERAL_ID    DT_ATMEL_SAM_USART_400A4000_PERIPHERAL_ID
```
Comment (mnkp) [inline soc/arm/atmel_sam/sam4e/dts_fixup.h]: This is probably unintentional?

**Review by mnkp — COMMENTED** [review]
Looks good, though I believe it is too early to convert all remaining SAM drivers to DT_INST. This family requires pinctrl data to be provided by DTS. For now we could convert watchdog, as this PR proposes, GPIO and counter. Any driver that doesn't require pinctlr.

File: soc/arm/atmel_sam/sam4e/dts_fixup.h (hunk 1 above)
Comment (galak) [reply soc/arm/atmel_sam/sam4e/dts_fixup.h]: It was a merge issue, good catch.

**Comment by galak** [comment]
> Looks good, though I believe it is too early to convert all remaining SAM drivers to DT_INST. This family requires pinctrl data to be provided by DTS. For now we could convert watchdog, as this PR proposes, GPIO and counter. Any driver that doesn't require pinctlr.

Going to handle this with a mix of DT_INST and DT_NODELABEL.  Once NODELABEL is in place I'll work up a PR to show the mix of pinctrl + driver.

**Review by mnkp — APPROVED** [review]
I haven't done full review before, I was under impression that more drivers will be added to this PR. The two commits converting watchdog and entropy drivers look good.

**Comment by galak** [comment]
> I haven't done full review before, I was under impression that more drivers will be added to this PR. The two commits converting watchdog and entropy drivers look good.

I was thinking its best to merge and close out the PRs for the easy stuff and than have follow on PRs.  Was thinking this was going to go faster, and thus the previous model that you are thinking about.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**